executorlib/base/executor.py

```
@@ -0,0 +1,148 @@
+
+    @property
+    def future_queue(self) -> queue.Queue:
+        """
+        Get the future queue.
+
+        Returns:
+            queue.Queue: The future queue.
+        """
+        return self._future_queue
+
+    def submit(self, fn: callable, *args, resource_dict: dict = {}, **kwargs) -> Future:
```

Ruff

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)